app/controllers/ems_cluster_controller.rb

```diff
@@ -23,6 +23,14 @@ def self.display_methods
     %w[all_vms miq_templates vms hosts resource_pools config_info storage custom_button_events]
   end
 
+  def record_class
+    if %w[all_vms vms].include?(params[:display]) || params[:pressed].starts_with?('miq_template')
+      VmOrTemplate
```

Comment:
You could call `super` here if you define a *default* `record_class` in the `CiProcessing`.